waiter/src/waiter/util/async_utils.clj

```diff
@@ -22,6 +22,11 @@
   (:import (clojure.core.async.impl.channels ManyToManyChannel)
            (java.util.concurrent ExecutorService)))
 
+(defn singleton-chan [v]
```

Comment:
Nitpick: we  could use a `promise-chan`?

Reply:
OK, yeah, I checked and a promise-chan actually uses a special buffer for the single value. Switched.